src/panels/lovelace/cards/hui-entities-card.ts

```diff
@@ -0,0 +1,191 @@
+    return {
+      config: {},
+    };
+  }
+
+  public getCardSize() {
+    if (!this.config) {
+      return 0;
+    }
+    // +1 for the header
+    return (this.config.title ? 1 : 0) + this.config.entities.length;
+  }
+
+  public setConfig(config: Config) {
+    const entities = processConfigEntities(config.entities);
+    for (const entity of entities) {
+      if (
+        entity.type === "call-service" &&
+        (!entity.service ||
+          !entity.name ||
+          !entity.icon ||
+          !entity.service_data ||
+          !entity.action_name)
+      ) {
+        throw new Error("Missing required property when type is call-service");
+      } else if (
+        entity.type === "weblink" &&
+        (!entity.name || !entity.icon || !entity.url)
+      ) {
+        throw new Error("Missing required property when type is weblink");
+      }
+    }
+
+    this.config = config;
+    this.configEntities = entities;
+  }
+
+  protected render() {
+    if (!this.config || !this._hass) {
+      return html``;
+    }
+    const { show_header_toggle, title } = this.config;
+
+    return html`
+      ${this.renderStyle()}
+      <ha-card>
+        ${
+          !title && !show_header_toggle
+            ? html``
+            : html`
+            <div class='header'>
+              <div class="name">${title}</div>
+              ${
+                show_header_toggle === false
+                  ? html``
+                  : html`
+                  <hui-entities-toggle
+                    .hass="${this._hass}"
+                    .entities="${this.configEntities!.map(
+                      (conf) => conf.entity
+                    )}"
+                  >
+                  </hui-entities-toggle>`
+              }
+            </div>`
+        }
+        <div id="states">
+          ${this.configEntities!.map((entityConf) =>
+            this.renderEntity(entityConf)
+          )}
+        </div>
+      </ha-card>
+    `;
+  }
+
+  private renderStyle() {
+    return html`
+      <style>
+        ha-card {
+          padding: 16px;
+        }
+        #states {
+          margin: -4px 0;
+        }
+        #states > * {
+          margin: 4px 0;
+        }
+        #states > * {
+          overflow: hidden;
+        }
+        .header {
+          @apply --paper-font-headline;
+          /* overwriting line-height +8 because entity-toggle can be 40px height,
+            compensating this with reduced padding */
+          line-height: 40px;
+          color: var(--primary-text-color);
+          padding: 4px 0 12px;
+          display: flex;
+          justify-content: space-between;
+        }
+        .header .name {
+          @apply --paper-font-common-nowrap;
+        }
+        .state-card-dialog {
+          cursor: pointer;
+        }
+      </style>
+    `;
+  }
+
+  private renderEntity(entityConf) {
+    const element = createRowElement(entityConf);
+    element.hass = this._hass;
+    element.entityConf = entityConf;
```

Comment:
We should not set random properties on web-components (on divs it's fine). In this case I would change the click lister:

```ts
element.addEventListener("click", (ev: MouseEvent) => this._handleClick(entityConf));
```

Reply:
Imagine a entity row component having their own entityConf and we get random bugs 😱